cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/MonitorHandler.java

```diff
@@ -84,6 +96,9 @@ public void getServiceLiveInfo(HttpRequest request, HttpResponder responder,
       throw new NotFoundException(String.format("Invalid service name %s", serviceName));
     }
 
+    SystemServiceId systemServiceId = new SystemServiceId(serviceName);
+    accessEnforcer.enforce(systemServiceId, authenticationContext.getPrincipal(), StandardPermission.GET);
```

Comment:
use `ContextAccessEnforcer` instead of this two

Reply:
Done.